### ISSUE TITLE
Verify "Host" in Http header

### ISSUE THREAD

**Issue by jimpetersson** [rschildmeijer/deft#25]
Because of the urgency of implementing the new Host: header, servers are not allowed to tolerate HTTP 1.1 requests without it. If a server receives such a request, it must return a "400 Bad Request" response, like

```
HTTP/1.1 400 Bad Request
Content-Type: text/html
Content-Length: 111

<html><body>
<h2>No Host: header received</h2>
HTTP 1.1 requests must include the Host: header.
</body></html>
```

This requirement applies only to clients using HTTP 1.1, not any future version of HTTP. If the request uses an HTTP version later than 1.1, the server can accept an absolute URL instead of a Host: header (see next section). If the request uses HTTP 1.0, the server may accept the request without any host identification. 

See: http://www.jmarshall.com/easy/http/

**Comment by rschildmeijer** [comment]
Lame http rule. We could add this check to HttpRequest.of and do appropriate action (throw exception?) when Host header is missing for 1.1 http requests

**Comment by jimpetersson** [comment]
Yes, lame. But what to do :)
Maybe better to have HttpProtocolImpl.handleRead() call a "verify" method, which verifies that the request is ok?

**Comment by rschildmeijer** [comment]
then we need to parse the "http request data" twice

**Comment by jimpetersson** [comment]
HTTP 1.0 defines 16 headers, though none are required. HTTP 1.1 defines 46 headers, and one (Host:) is required in requests.

**Comment by jimpetersson** [comment]
Added verification of HttpRequest. If version = HTTP/1.1 the Host header is required. Closed by 03b99335333a76a73d3c3c7a0eeeabc81e9d8d30
